packages/page-calendar/src/DayItem.tsx

```diff
@@ -46,62 +50,116 @@ function DayItem ({ className, item: { date, info, type } }: Props): React.React
+          break;
 
         case 'democracyLaunch':
-          return <><div className='itemDesc'>{t<string>('Start of the next referendum voting period')}</div>{typeLink}</>;
+          s = 'Start of the next referendum voting period';
+          break;
 
         case 'referendumDispatch':
-          return <><div className='itemDesc'>{t<string>('Potential dispatch of referendum {{id}} (if passed)', { replace: { id } })}</div>{typeLink}</>;
+          s = 'Potential dispatch of referendum {{id}} (if passed)';
+          break;
 
         case 'referendumVote':
-          return <><div className='itemDesc'>{t<string>('Voting ends for referendum {{id}}', { replace: { id } })}</div>{typeLink}</>;
+          s = 'Voting ends for referendum {{id}}';
+          break;
 
         case 'scheduler':
-          return <><div className='itemDesc'>{
-            id
-              ? t<string>('Execute named scheduled task {{id}}', { replace: { id } })
-              : t<string>('Execute anonymous scheduled task')
-          }</div>{typeLink}</>;
+          s = 'Execute named scheduled task';
+          id && (s = s + ' {{id}}');
+          break;
 
         case 'stakingEpoch':
-          return <><div className='itemDesc'>{t<string>('Start of a new staking session {{id}}', { replace: { id } })}</div>{typeLink}</>;
+          s = 'Start of a new staking session {{id}}';
+          break;
 
         case 'stakingEra':
-          return <><div className='itemDesc'>{t<string>('Start of a new staking era {{id}}', { replace: { id } })}</div>{typeLink}</>;
+          s = 'Start of a new staking era {{id}}';
+          break;
 
         case 'stakingSlash':
-          return <><div className='itemDesc'>{t<string>('Application of slashes from era {{id}}', { replace: { id } })}</div>{typeLink}</>;
+          s = 'Application of slashes from era {{id}}';
+          break;
 
         case 'treasurySpend':
-          return <><div className='itemDesc'>{t<string>('Start of next spending period')}</div>{typeLink}</>;
+          s = 'Start of next spending period';
+          break;
 
         case 'societyChallenge':
-          return <><div className='itemDesc'>{t<string>('Start of next membership challenge period')}</div>{typeLink}</>;
+          s = 'Start of next membership challenge period';
+          break;
 
         case 'societyRotate':
-          return <><div className='itemDesc'>{t<string>('Acceptance of new members and bids')}</div>{typeLink}</>;
+          s = 'Acceptance of new members and bids';
+          break;
 
         default:
           return assertUnreachable(type);
       }
+
+      setDescription(id ? s.replaceAll('{{id}}', id) : s);
+
+      return id ? <><div className='itemDesc'>{t<string>(s, { replace: { id } })}</div>{typeLink}</> : (<><div className='itemDesc'>{s}</div>{typeLink}</>);
     },
     [info, t, type]
   );
 
+  function exportToCal (fileName: string, date: Date): void {
```

Comment:
Extract this function to file-level - as it stands it gets re-created each render.